File: app/src/main/java/net/mmho/photomap2/PhotoCardLayout.java
package net.mmho.photomap2;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;
import android.widget.TextView;


public class PhotoCardLayout extends RelativeLayout {

    private ThumbnailImageView thumbnail;
    private TextView count;

    public PhotoCardLayout(Context context) {
        this(context,null);
    }

    public PhotoCardLayout(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public PhotoCardLayout(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        thumbnail = (ThumbnailImageView)findViewById(R.id.thumbnail);
        count = (TextView)findViewById(R.id.count);
    }

    public void setData(PhotoGroup g){
        count.setText(String.format("%2d",g.size()));
        thumbnail.startLoading(g.get(0).getPhotoId());
    }

}

File: app/src/main/kotlin/net/mmho/photomap2/PhotoCardLayout.kt
package net.mmho.photomap2

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import kotlinx.android.synthetic.main.layout_photo_card.view.*


class PhotoCardLayout @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
        : RelativeLayout(context, attrs, defStyle) {

    fun setData(g: PhotoGroup) {
        @SuppressLint("SetTextI18n")
        count.text = "%2d".format(g.size)
        thumbnail.load(g[0].photo_id)
    }

}
